docs/design/CONFIG_API.md

```diff
@@ -0,0 +1,228 @@
+	//
+	// https://gateway-api.sigs.k8s.io/v1alpha2/references/spec/#gateway.networking.k8s.io/v1alpha2.GatewayClass
+	//
+	// +optional
+	controllerName string `json:"controllerName,omitempty"`
+}
+
+// Providers define desired configuration of Envoy Gateway providers.
+type Providers struct {
+	// Kubernetes defines the configuration of the Kubernetes provider. Kubernetes
+	// provides runtime configuration via the Kubernetes API.
+	//
+	// +support:alpha
+	// +optional
+	Kubernetes *kubernetes.Provider `json:"kubernetes,omitempty"`
+
+	// File defines the configuration of the File provider. File provides runtime
+	// configuration defined by one or more files.
+	//
+	// +support:alpha
+	// +optional
+	File *file.Provider `json:"file,omitempty"`
+}
+
+// XdsServer defines the desired configuration of the Envoy Gateway xDS server.
+type XdsServer struct {
+	// Defines the IP address for server to serve xDS over gRPC.
+	//
+	// If unspecified, defaults to "0.0.0.0", e.g. all IP addresses.
+	//
+	// +optional
+	Address *string `json:"address,omitempty"`
+
+	// Defines the network port for the xDS server to serve xDS over gRPC.
+	//
+	// If unspecified, defaults is 18000.
+	//
+	// +optional
+	Port *int32 `json:"port,omitempty"`
+}
+```
+Note that a provider-specific configuration is defined in the provider package. The following is an example of the
+Kubernetes provider:
+```go
+// gateway/internal/kubernetes/kubernetes.go
+
+package kubernetes
+
+// Provider defines the configuration of the Kubernetes provider.
+type Provider struct {
+	// TODO: Define Kubernetes configuration fields, e.g. restrict namespaces to watch
+	//       Gateway/HTTPRoute resources.
+}
+```
+
+### Gateway
+Gateway defines desired configuration of [Gateway API][gw_api] controllers that reconcile and translate Gateway API
+resources into the Intermediate Representation (IR). Refer to the Envoy Gateway [design doc][design_doc] for additional
+details.
+
+### Providers
+Providers define desired configuration of Envoy Gateway providers. Providers are infrastructure components that Envoy
+Gateway calls to establish its runtime configuration. Providers are defined by a `+support` marker to indicate the
+stability of the provider. Refer to the Envoy Gateway [design doc][design_doc] for additional details.
+
+### XdsServer
+XdsServer defines desired configuration of [Xds][xds] server configuration parameters. Refer to the Envoy Gateway
+[design doc][design_doc] for additional details.
+
+### Configuration Example
+The configuration file is defined by the ControlPlaneSpec API type. At startup, Envoy Gateway searches for the
+configuration at "/etc/envoy-gateway/config.yaml".
+
+Start Envoy Gateway:
+```shell
+envoy-gateway start
+```
+Since the configuration file does not exist, Envoy Gateway will start with default configuration parameters.
+
+The Kubernetes provider can be configured explicitly using `providers.kubernetes`:
+```yaml
+$ cat << EOF > /etc/envoy-gateway/config.yaml
```

Comment:
the spec here should encapsulate `ControlPlaneSpec` rather than be `ControlPlaneSpec` i.e.
this allows for further extensions such as adding `dataPlane` here or something else
```
controlPlane:
  providers:
    kubernetes: {}
 ```

Reply:
@arkodg how would EG config 3rd party components (aux control planes in the design doc)? For example, run EG on Kubernetes with the [Rate Limit](https://github.com/envoyproxy/ratelimit) service?

Reply:
I updated the PR based on ^ comments.